packages/mwp-app-render/src/components/Dom.test.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Dom from './Dom';
+
+const MOCK_BASE_URL = 'https://www.mock-base-url.com';  
+
+const MOCK_PROPS = {
+	appMarkup: '<div>mock app markup</div>',
+	assetPublicPath: '/mock/asset/public/path',
+	baseUrl: MOCK_BASE_URL,
+	head: {
+		title: { toComponent: () => null },
+		meta: { toComponent: () => null },
+		link: { toComponent: () => null },
+		script: { toComponent: () => null },
+	},
+	initialState: {
+		config: {
+			apiUrl: 'https://www.mock-api-url.com',
+			baseUrl: MOCK_BASE_URL,
+			enableServiceWorker: true,
+			requestLanguage: 'fr-FR',
+			supportedLangs: ['en-US', 'de-De', 'es-ES'],
+			initialNow: '1517589465414',
+			variants: 'mock-variants',
+			entryPath: '/ny-tech',
+			media: 'mobile',
+		},
+	}, 
+	scripts: ['script.js'],
+	cssLinks: ['style.css'],
+};
+
+describe('Dom', function() {
+	const domComponent = shallow(
```

Comment:
nitpick, but in general it's helpful to make unit tests (each `it()` call) as self-contained as possible - i.e. avoid relying on parent scope so that maintainers don't have to search around the file to determine what is actually being tested. You don't have to optimize for DRYness. In this case, prefer calling `shallow` inside the `it()` - you can even do it inline in the `expect()` call without losing much clarity

```js
expect(shallow(<Dom {...MOCK_PROPS} />)).toMatchSnapshot();
```

Reply:
First, I'm happy to change this!

---

As a more general matter, I'm wondering if this feedback (regarding being self-contained) is applicable to this nearby file as well?

https://github.com/meetup/meetup-web-platform/blob/master/packages/mwp-app-render/src/components/ServerApp.test.jsx

FWIW, I happened to take inspiration from this file before submitting. I know it's a nitpick to begin with and my test file isn't exactly the same (`MOCK_PROPS`, for instance). However, I think this might be an okay opportunity to say that I have experienced this before -- where a particular feedback in PR _feels_ somewhat at odds with pre-existing code.

On the one hand -- I don't see this as a big deal. It's trivial for me to make this change and move on. On the other, I wonder if this is worth mentioning in case we can make improvements somewhere along the line. For me, I think it would be great if newcomers could feel like they could reliably follow existing patterns from the codebase.

Anyway, just some unfiltered feedback that came to mind. Happy to change this.

Reply:
Fair point. Given the size of our codebase and how quickly it grows, we won't ever be able to have 100% consistency on things like this - best practices evolve and we can't spend time refactoring old code to match, so although it's good to use existing code as a reference, it shouldn't ever be hallowed as an 'ideal'.

Usually when I preface with 'nitpick', it's because I'm going to make a non-blocking style/structure suggestion that you may also find value in - if you think it's _not_ useful, feel free to ignore it. It's mainly a way of sharing ideas about how to do things better, but in a subjective way.

The existing `ServerApp.test.jsx` test could definitely have been structured better, but it's not a stop-the-world sort of concern.

Reply:
Yep yep @mmcgahan, I'm with you on all that. And I definitely don't mind the nitpicks, it's more that I wanted to take this opportunity to be reflective about something I've witnessed.